### ISSUE TITLE
File opens too long if big folder is present on same level

### ISSUE THREAD

**Issue by vimmerru** [btrask/Sequential#9]
Steps to reproduce:
1. Put few images to desktop  (or any another folder)
2. Put big folder or smilink to big folder near. For example i have link to documents folder on my desktop (10k+ photos in subtree)
3. Open one of images in Sequential

Expected result:
- File opened in few seconds

Actual result:
- I see waiting cursor for minutes and have to kill Sequential

**Comment by btrask** [comment]
Yes, this is a known issue. There are two possible solutions:
1. Make opening large directories faster and less obtrusive
2. Only open the individual file

I don't know if or when I will have time to work on this, but change 2 shouldn't be that difficult.

Sorry for the inconvenience.

**Comment by pskowronek** [comment]
Yes, this is very annoying - especially when you want to see an image that is saved in /tmp, or ~/Desktop and you have lots of heavy folders there. Then the app gets frozen for a pretty long time until it processes all the folders - sometimes I had to kill the app, because I didn't want to wait that long.

I would say that, when the app is started:
- a main thread should open a chosen file immediately 
- a background thread should be started to:
  -- firstly scan the folder where the file is present and probably another thread to generate previews
  -- secondly scan subdirectories up to the first levels only w/o preview generation, plus a deeper scanning seems unnecessary til user navigates to a subfolder
- there should be a spinning wheel /progress indicator/ telling that the scanning is being made with an option to pause the background thread
- should be possible to close the app w/o waiting for the background thread to complete its task

Just my two cents :)

**Comment by tom--** [comment]
The issue is such that Sequential isn't useful for me.

Solution 2. that @btrask mentioned is fine.

For something more responsive, the name of the app "Sequential" says what: While the user is viewing the opened file, load the next and previous so either can be displayed immediately when the user moves forwards or back. kis
